server/src/main/java/org/opensearch/action/get/TransportGetAction.java

```diff
@@ -90,14 +93,21 @@ protected boolean resolveIndex(GetRequest request) {
 
     @Override
     protected ShardIterator shards(ClusterState state, InternalRequest request) {
+        String preference = request.request().preference();
+        // route realtime GET requests when segment replication is enabled to primary shards,
+        // iff there are no other preferences/routings enabled for routing to a specific shard
+        if (state.getMetadata()
```

Comment:
Lets pls move this to a private function for readability?

Something like `isPrimaryBasedRouting`